### PR TITLE
Implement Binary Tree to Binary Search Tree Conversion - Solution

### PR DESCRIPTION
**Pull request by k88manish**
@ashokdey @TheSTL - Raised a new PR because the last PR was raised with the wrong Author information

### CODE REVIEW

File: src/_Problems_/binary-tree-to-binary-search-tree/binary-tree-to-binary-search-tree.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('Binary tree to binary search tree', () => {
+  let tree;
+
+  describe('Create Binary Tree', () => {
+    tree = new Node(10);
+    tree.leftChild = new Node(30);
+    tree.leftChild.leftChild = new Node(20);
+    tree.rightChild = new Node(15);
+    tree.rightChild.rightChild = new Node(5);
```
Comment (TheSTL) [inline src/_Problems_/binary-tree-to-binary-search-tree/binary-tree-to-binary-search-tree.test.js]: [Binary Tree](https://github.com/knaxus/problem-solving-javascript/blob/master/src/_DataStructures_/Trees/BinaryTree/index.js) implementation is already present.

File: src/_Problems_/binary-tree-to-binary-search-tree/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  // Finally update the right subtree
+  arrayToBST(arr, node.rightChild);
+}
+
+function binaryTreeToBST(root) {
+  // Tree is empty
+  if (!root) return;
+  const arr = storeInorder(root);
+  arr.sort((a, b) => a - b);
+  arrayToBST(arr, root);
```
[the remark this answers is not in the document]
Comment (TheSTL) [reply src/_Problems_/binary-tree-to-binary-search-tree/index.js]: @ashokdey  Yes, it should return the root node of new created BST.

File: src/_Problems_/binary-tree-to-binary-search-tree/binary-tree-to-binary-search-tree.test.js (hunk 1 above)
Comment (ashokdey) [reply src/_Problems_/binary-tree-to-binary-search-tree/binary-tree-to-binary-search-tree.test.js]: @k88manish, you can use this [Binary Tree using level order](https://github.com/knaxus/problem-solving-javascript/blob/master/src/_DataStructures_/Trees/BinaryTree) to create the tree here.

**Comment by k88manish** [comment]
Hi @ashokdey @TheSTL , I have made changes as per the review and also made a correction for the binary search file as it was creating child nodes for null value as well.

**Comment by ashokdey** [comment]
@k88manish, ok for me. waiting for @TheSTL

**Pull request merged**